pepys_import/core/store/data_store.py

```diff
@@ -1647,3 +1647,99 @@ def get_logs_by_change_id(self, change_id):
             .filter(self.db_classes.Log.change_id == change_id)
             .all()
         )
+
+    def merge_platforms(self, platform_list, master_id) -> bool:
+        """Merges given platforms. Moves sensors from other platforms to the Target platform.
+        If sensor with same name is already present on Target platform, moves measurements
+        to that sensor. Also moves all comments from other platforms to the Target platform.
+
+        :param platform_list: A list of platform IDs or platform objects
+        :type platform_list: List
+        :param master_id: Target platform's ID or objects itself
+        :type master_id: UUID or Platform
+        :return: True if merging completed successfully, False otherwise.
+        :rtype: bool
+        """
+        Platform = self.db_classes.Platform
+        Sensor = self.db_classes.Sensor
+        State = self.db_classes.State
+        Contact = self.db_classes.Contact
+
+        if isinstance(master_id, Platform):
+            master_id = master_id.platform_id
+
+        master_platform = (
+            self.session.query(Platform).filter(Platform.platform_id == master_id).scalar()
+        )
+        if not master_platform:
+            print(f"No platform found with the given master_id: '{master_id}'!")
```

Comment:
As this is going to be called from within a GUI, could you change this (and any other error messages below) to raise an exception instead? A `ValueError` with an appropriate error string would work fine here, and could be caught by the GUI to display the error nicely (the standard print statement doesn't play nicely with prompt_toolkit's GUIs, and will break various things...)

Reply:
Okay, I changed it: https://github.com/debrief/pepys-import/pull/740/commits/94ce9d8240683472553fd09c225987e45e4f70b3